client/block-builder/src/lib.rs

```diff
@@ -206,21 +213,22 @@ where
 	/// The storage proof will be `Some(_)` when proof recording was enabled.
 	pub fn build(
 		mut self,
-	) -> Result<BuiltBlock<Block, backend::StateBackendFor<B, Block>>, ApiErrorFor<A, Block>> {
+		context_provider: &dyn Fn(H256) -> Option<(AccountId32, u32)>
```

Comment:
totally missing where the `context_provider` is coming from, can you explain where it is implemented?